webapp/exercise_admin/views.py

```diff
@@ -405,17 +405,44 @@ def get_instance_files(request, exercise):
     
     for instance_file in instance_files:
         try:
-            link = InstanceIncludeFileToExerciseLink.objects.get(include_file=instance_file, exercise=exercise)
+            link = InstanceIncludeFileToExerciseLink.objects.get(include_file=instance_file, exercise=exercise_id)
         except InstanceIncludeFileToExerciseLink.DoesNotExist:
             link = None
         instance_file_json = {
+            "id" : instance_file.id,
             "default_name": instance_file.default_name,
             "description" : instance_file.description,
-            "link" : link,
+            "link" : {
+                "id" : link.id,
+                "name" : link.file_settings.name,
+                "purpose" : link.file_settings.purpose,
+                "chown_settings" : link.file_settings.chown_settings,
+                "chgrp_settings" : link.file_settings.chgrp_settings,
+                "chmod_settings" : link.file_settings.chmod_settings,
+            }
```

Comment:
The fields that have translations should be referenced to with the proper _{language_code} prefixes. Otherwise only the currently set site wide translation preference will be used.